tests/scenarios/delegated-grant.spec.ts

```diff
@@ -641,9 +759,83 @@ export function testDelegatedGrantScenarios(): void {
         }
       });
 
-      const deviceXWriteReply = await dwn.processMessage(bob.did, recordsReadByDeviceX.message);
-      expect(deviceXWriteReply.status.code).to.equal(401);
-      expect(deviceXWriteReply.status.detail).to.contain(DwnErrorCode.RecordsGrantAuthorizationScopeProtocolMismatch);
+      const deviceXReadReply = await dwn.processMessage(bob.did, recordsReadByDeviceX.message);
+      expect(deviceXReadReply.status.code).to.equal(401);
+      expect(deviceXReadReply.status.detail).to.contain(DwnErrorCode.RecordsGrantAuthorizationScopeProtocolMismatch);
+    });
+
+    it('should fail if delegated grant has a mismatching protocol scope - subscribe', async () => {
+      // scenario:
+      // 1. Alice creates a delegated grant for device X to act as her for a protocol that is NOT chat protocol
+      // 2. Bob starts a chat thread with Alice on his DWN
+      // 3. Device X attempts to use the delegated grant to subscribe to the thread
+      // 4. Bob's DWN should reject Device X's subscribe attempt
+
+      const alice = await DidKeyResolver.generate();
+      const deviceX = await DidKeyResolver.generate();
+      const bob = await DidKeyResolver.generate();
+
+      // Bob has the chat protocol installed
+      const protocolDefinition = threadRoleProtocolDefinition;
+      const protocol = threadRoleProtocolDefinition.protocol;
+      const protocolsConfig = await TestDataGenerator.generateProtocolsConfigure({
+        author: bob,
+        protocolDefinition
+      });
+      const protocolsConfigureReply = await dwn.processMessage(bob.did, protocolsConfig.message);
+      expect(protocolsConfigureReply.status.code).to.equal(202);
+
+      // Bob starts a chat thread
+      const threadRecord = await TestDataGenerator.generateRecordsWrite({
+        author       : bob,
+        protocol     : protocolDefinition.protocol,
+        protocolPath : 'thread',
+      });
+      const threadRoleReply = await dwn.processMessage(bob.did, threadRecord.message, { dataStream: threadRecord.dataStream });
+      expect(threadRoleReply.status.code).to.equal(202);
+
+      // Bob adds Alice as a participant in the thread
+      const participantRoleRecord = await TestDataGenerator.generateRecordsWrite({
+        author       : bob,
+        recipient    : alice.did,
+        protocol     : protocolDefinition.protocol,
+        protocolPath : 'thread/participant',
+        contextId    : threadRecord.message.contextId,
+        parentId     : threadRecord.message.recordId,
+        data         : new TextEncoder().encode('Alice is my friend'),
+      });
+      const participantRoleReply = await dwn.processMessage(bob.did, participantRoleRecord.message, { dataStream: participantRoleRecord.dataStream });
+      expect(participantRoleReply.status.code).to.equal(202);
```

Comment:
Consider:

1. Just include the subscribe test in the read & query like what you did. or
1. Removing this block and move the test into functional test file.

This block is basically irrelevant (thus can be removed by just setting the `contextId` to `irrelevant`), because authorization takes place first.

Giving more context: Read and delete needed the initial setup because the code path is different: they fetch the record of that real `recordId` first before authorization.

Query on the other hand require less setup, so I figured that I'd roll it into the read.

Reply:
Yup, makes sense.

I went with #1. See if you're ok with the order of events/commenting I put in place.